### PR TITLE
Use multi package project structure to build web and native clients

### PR DESCRIPTION
**Pull request by VikashKothary**
It's a nasty PR but works. Let me know how you want to progress merging it back into master.

We can also have a knowledge transfer on how to use `lerna` and `yarn workspaces` but all you need to do is run `yarn start` to run the create-react-app web client.

### CODE REVIEW

**Comment by rhys-saldanha** [comment]
I think the Netlify integration is broken, let me fix that first. You might need to add some config so that Netlify knowns how to build this and what to deploy

**Comment by rhys-saldanha** [comment]
```
>lerna --scope @reciplease/webclient run build
lerna ERR! yarn run build exited 1 in '@reciplease/webclient'
lerna ERR! yarn run build stdout:
yarn run v1.22.10
$ react-scripts build
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

lerna ERR! yarn run build stderr:

There might be a problem with the project dependency tree.
It is likely not a bug in Create React App, but something you need to fix locally.

The react-scripts package provided by Create React App requires a dependency:

  "eslint": "^6.6.0"

Don't try to install it manually: your package manager does it automatically.
However, a different version of eslint was detected higher up in the tree:

  C:\git\reciplease-web\node_modules\eslint (version: 7.11.0)

Manually installing incompatible versions is known to cause hard-to-debug issues.

If you would prefer to ignore this check, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That will permanently disable this message but you might encounter other issues.

To fix the dependency tree, try following the steps below in the exact order:

  1. Delete package-lock.json (not package.json!) and/or yarn.lock in your project folder.
  2. Delete node_modules in your project folder.
  3. Remove "eslint" from dependencies and/or devDependencies in the package.json file in your project folder.
  4. Run npm install or yarn, depending on the package manager you use.

In most cases, this should be enough to fix the problem.
If this has not helped, there are a few other things you can try:

  5. If you used npm, install yarn (http://yarnpkg.com/) and repeat the above steps with it instead.
     This may help because npm has known issues with package hoisting which may get resolved in future versions.

  6. Check if C:\git\reciplease-web\node_modules\eslint is outside your project directory.
     For example, you might have accidentally installed something in your home folder.

  7. Try running npm ls eslint in your project folder.
     This will tell you which other package (apart from the expected react-scripts) installed eslint.

If nothing else helps, add SKIP_PREFLIGHT_CHECK=true to an .env file in your project.
That would permanently disable this preflight check in case you want to proceed anyway.

P.S. We know this message is long but please read the steps above :-) We hope you find them helpful!

error Command failed with exit code 1.

lerna ERR! yarn run build exited 1 in '@reciplease/webclient'
```

**Comment by VikashKothary** [comment]
As yes. The problem exists because the `create-react-app` dependencies are conflicting with the installed versions. The installed versions are newer so it should be fine but you have a few options:

1) Set SKIP_PREFLIGHT_CHECK=true. That's what I've been doing during development and everything works.

2) Use `yarn resolutions` to override the versions used in `create-react-app`. Easy but you have to do it one by one.

3) Downgrade the versions used in the project to match the versions used in `create-react-app`. Easy but you can't use the latest versions.

4) Use separate `node_modules` for `create-react-app`. Easy but duplicated dependencies.

This hasn't been fixed because we were originally going to remove `create-react-app` in favour of `react-native-web`. Then I thought there would be value to have both. One which will be the web app, and another, will be the web site for the app. 

Does that makes sense? Thoughts?

**Comment by rhys-saldanha** [comment]
(5) didn't work, I needed to change the webclient build command to `cross-env SKIP_PREFLIGHT_CHECK=true react-scripts build`. I still think that is this is too much of a complication this early in the project. Plus, I can't seem to even build the original website anymore...
We can revisit separate component, web and native package at a later date, for now I think we need to continue making the website to better understand what user flows we have / will have. Once that's more or less done, it's not impossible to refactor, pull out common components, molecules, and pages for a dual web and native deploy.

File: packages/nativeclient/App.tsx [hunk 1]
```diff
@@ -0,0 +1,29 @@
+import { StatusBar } from 'expo-status-bar';
+import React from 'react';
+import { SafeAreaProvider } from 'react-native-safe-area-context';
+import { Text } from 'react-native';
+
+// import useCachedResources from './hooks/useCachedResources';
```
Comment (VikashKothary) [inline packages/nativeclient/App.tsx:6]: Remove commented code

**Pull request closed without merging**